parts/k8s/windowskubeletfunc.ps1

```diff
@@ -130,30 +130,39 @@ users:
     $kubeConfig | Out-File -encoding ASCII -filepath "$kubeConfigFile"
 }
 
+function
+Build-PauseContainer {
+    Param(
+        [Parameter(Mandatory = $true)][string]
+        $WindowsBase,
+        $DestinationTag
+    )
+    # Future work: This needs to build wincat - see https://github.com/Azure/aks-engine/issues/1461
+    "FROM $($WindowsBase)" | Out-File -encoding ascii -FilePath Dockerfile
+    "CMD cmd /c ping -t localhost" | Out-File -encoding ascii -FilePath Dockerfile -Append
+    docker build -t $containerTag .
+    return $containerTag
+}
+
 function
 New-InfraContainer {
     Param(
         [Parameter(Mandatory = $true)][string]
-        $KubeDir
+        $KubeDir,
+        $DestinationTag = "kubletwin/pause"
     )
     cd $KubeDir
     $computerInfo = Get-ComputerInfo
-    $windowsBase = if ($computerInfo.WindowsVersion -eq "1709") {
-        "microsoft/nanoserver:1709"
-    }
-    elseif ($computerInfo.WindowsVersion -eq "1803") {
-        "microsoft/nanoserver:1803"
-    }
-    elseif ($computerInfo.WindowsVersion -eq "1809") {
-        "mcr.microsoft.com/windows/nanoserver:1809"
-    }
-    else {
-        "mcr.microsoft.com/nanoserver-insider"
-    }
 
-    "FROM $($windowsBase)" | Out-File -encoding ascii -FilePath Dockerfile
-    "CMD cmd /c ping -t localhost" | Out-File -encoding ascii -FilePath Dockerfile -Append
-    docker build -t kubletwin/pause .
+    # Reference for these tags: curl -L https://mcr.microsoft.com/v2/k8s/core/pause/tags/list
+    # Then docker run --rm mplatform/manifest-tool inspect mcr.microsoft.com/k8s/core/pause:<tag>
+
+    $windowsBase = switch ($computerInfo.WindowsVersion) {

```

Comment:
I'm debating whether or not this version-specific behavior should be in scripts on the node, or part of AKS-Engine.

Pros for AKS-Engine:

- can override pause image with another one in ACR, Docker Hub, or elsewhere

Pros for this script:

- AKS-Engine could deploy multiple Windows agent pools and each node will automatically figure out the right image